L/LibCURL/build_tarballs.jl

```diff
@@ -15,6 +15,12 @@ sources = [
 script = raw"""
 cd $WORKSPACE/srcdir/curl-*
 
+# Work around missing library:
+# `Error loading shared library libmbedcrypto.so.5: No such file or directory (needed by /workspace/destdir/lib/libssh2.so.1)`
+if [[ ${target} == x86_64-linux-musl ]]; then
+   ln -s libmbedcrypto.so ${WORKSPACE}/destdir/lib/libmbedcrypto.so.5
+fi
```

Comment:
This isn't a good solution: the symlink remains there at the end of the build and ends up in the tarball, leaving a [broken symlink](https://dev.azure.com/JuliaPackaging/Yggdrasil/_build/results?buildId=8787&view=logs&j=a6122f81-e5a2-5211-451a-3f540ebd06e0&t=ff55cc42-c3ce-53e5-62b6-7d1c7f90b9e0&l=5003).  I haven't looked into the issue, but what happens usually is that some host libraries are picked up (in `/usr/lib`).  What I do in those cases is to delete those :sweat_smile:

Reply:
I'm not a fan of this solution either. I figured the build environment wasn't setup correctly for LibCURL's runtime test. I'll look into `/usr/lib`

Reply:
I'm not seeing any other `libssh2` or `libmbed*` libraries on the host system that could be removed to address the issue.

Reply:
When building LibSSH2:
```
# ldd libssh2.so.1.0.1
	ldd (0x7f86a6067000)
	libmbedtls.so.13 => /workspace/destdir/lib/libmbedtls.so.13 (0x7f86a5bea000)
	libmbedx509.so.1 => /workspace/destdir/lib/libmbedx509.so.1 (0x7f86a59c3000)
	libmbedcrypto.so.6 => /workspace/destdir/lib/libmbedcrypto.so.6 (0x7f86a573d000)
	libc.musl-x86_64.so.1 => ldd (0x7f86a6067000)
```
When building LibCURL:
```
# ldd libssh2.so.1.0.1
	ldd (0x7fa9284fa000)
	libmbedtls.so.13 => /workspace/destdir/lib/libmbedtls.so.13 (0x7fa928074000)
	libmbedx509.so.1 => /workspace/destdir/lib/libmbedx509.so.1 (0x7fa927e4d000)
Error loading shared library libmbedcrypto.so.5: No such file or directory (needed by libssh2.so.1.0.1)
	libc.musl-x86_64.so.1 => ldd (0x7fa9284fa000)
	libmbedcrypto.so.6 => /workspace/destdir/lib/libmbedcrypto.so.6 (0x7fa927bc7000)
```

Reply:
@staticfloat once again has all the answers :smile:: https://github.com/JuliaPackaging/Yggdrasil/pull/2420